lib/string.js

```diff
@@ -430,4 +437,31 @@ internals.String.prototype.trim = function () {
     return obj;
 };
 
+
+internals.String.prototype.replace = function (pattern, replacement) {
+
+    Hoek.assert(pattern instanceof RegExp, 'pattern must be a RegExp');
+    Hoek.assert(typeof replacement === 'string', 'replacement must be a String');
+
+    var obj = this._test('replace', undefined, function (value, state, options) {
```

Comment:
I don't consider this to be a test like trim, you can't "reject" anything from this rule, this is mainly internal stuff, so just clone the current string.